contracts/pcv/utils/ERC20Allocator.sol

```diff
@@ -0,0 +1,459 @@
+    /// @param amount of token to be sent
+    function sweep(
+        address token,
+        address to,
+        uint256 amount
+    ) external onlyGovernor {
+        IERC20(token).safeTransfer(to, amount);
+    }
+
+    /// ----------- Permissionless PCV Allocation APIs -----------
+
+    /// @notice pull ERC20 tokens from PSM and send to PCV Deposit
+    /// if the amount of tokens held in the PSM is above
+    /// the target balance.
+    function skim(address pcvDeposit) external whenNotPaused {
+        address psm = pcvDepositToPSM[pcvDeposit];
+        _skim(psm, pcvDeposit);
+    }
+
+    /// @notice pull ERC20 tokens from PSM and send to PCV Deposit
+    /// if the amount of tokens held in the PSM is above
+    /// the target balance.
+    /// @param psm the peg stability to pull excess funds from
+    /// @param pcvDeposit the PCV Deposit to send excess funds to
+    function skim(address psm, address pcvDeposit) external whenNotPaused {
+        /// check that pcvDeposit has been whitelisted
+        /// will be true if both are address 0, but then check skim condition will fail
+        /// meaning we do not have to check if address is 0 here
+        require(
+            pcvDepositToPSM[pcvDeposit] == psm,
+            "ERC20Allocator: invalid pcvDeposit"
+        );
+        _skim(psm, pcvDeposit);
+    }
+
+    function _skim(address psm, address pcvDeposit) internal {
+        /// Check
+
+        /// note this check is redundant, as calculating amountToPull will revert
+        /// if pullThreshold is greater than the current balance of psm
+        /// however, we like to err on the side of verbosity
+        require(
+            _checkSkimCondition(psm),
+            "ERC20Allocator: skim condition not met"
+        );
+
+        (uint256 amountToSkim, uint256 adjustedAmountToSkim) = getSkimDetails(
+            pcvDeposit
+        );
+
+        /// Effects
+
+        _replenishBuffer(adjustedAmountToSkim);
+
+        /// Interactions
+
+        /// pull funds from pull target and send to push target
+        /// automatically pulls underlying token
+        PCVDeposit(psm).withdraw(pcvDeposit, amountToSkim);
+
+        /// deposit pulled funds into the selected yield venue
+        PCVDeposit(pcvDeposit).deposit();
+
+        emit Skimmed(amountToSkim);
+    }
+
+    /// @notice push ERC20 tokens to PSM by pulling from a PCV deposit
+    /// flow of funds: PCV Deposit -> PSM
+    /// @param pcvDeposit to pull funds from and send to corresponding PSM
+    function drip(address pcvDeposit) external whenNotPaused {
+        address psm = pcvDepositToPSM[pcvDeposit];
+        _drip(psm, PCVDeposit(pcvDeposit));
+    }
+
+    /// @notice push ERC20 tokens to PSM by pulling from a PCV deposit
+    /// flow of funds: PCV Deposit -> PSM
+    /// @param psm peg stability module to drip to
+    /// @param pcvDeposit pcv deposit to pull funds from
+    function drip(address psm, address pcvDeposit) external whenNotPaused {
```

Comment:
we can remove this function as psm address can always be inferred from the pcvDeposit address